server/webapp/WEB-INF/rails.new/spec/webpack/views/dashboard/pipeline_widget_spec.js

```diff
@@ -973,4 +986,26 @@ describe("Dashboard Pipeline Widget", () => {
     });
   }
 
+  function stubTriggerOptionsWithInvalidData(pipelineName) {
+    const response = "\n" +
+      "{\n" +
+      "  \"variables\" : [ ],\n" +
+      "  \"materials\" : [ {\n" +
+      "    \"type\" : \"Git\",\n" +
+      "    \"name\" : \"test-repo\",\n" +
+      "    \"fingerprint\" : \"4879d548de8a9d7122ceb71e7809c1f91a0876afa534a4f3ba7ed4a532bc1b02\",\n" +
+      "    \"revision\" : { : { : \"Failed due to an exception. Please check the logs.\"\n" +
+      "      }\n" +
+      "    }\n" +
+      "  } ]\n" +
+      "}";
+    jasmine.Ajax.stubRequest(`/go/api/pipelines/${pipelineName}/trigger_options`, undefined, 'GET').andReturn({
```

Comment:
No! No! No!

See https://github.com/gocd/gocd/blob/eb15ae0c1c4d68a02d217affed98466235536185/server/webapp/WEB-INF/rails.new/spec/webpack/views/roles/roles_widget_spec.js#L448

Reply:
This test is about a malformed JSON. `JSON.stringify` will not compile with an invalid JSON

Reply:
Ahh, makes sense. I was confused because I saw a huge json doc. Perhaps add a smaller json that visibly looks broken?